src/http/request_handler.rs

```diff
@@ -74,10 +74,12 @@ fn not_found() -> Result<Response<Body>> {
 fn create_request_context(req: &Request<Body>, app_ctx: &AppContext) -> RequestContext {
     let upstream = app_ctx.blueprint.upstream.clone();
     let allowed = upstream.allowed_headers;
-    let req_headers = create_allowed_headers(req.headers(), &allowed);
+    let allowed_headers = create_allowed_headers(req.headers(), &allowed);
 
     let _allowed = app_ctx.blueprint.server.get_experimental_headers();
-    RequestContext::from(app_ctx).request_headers(req_headers)
+    RequestContext::from(app_ctx)
+        .request_headers(req.headers().clone())
+        .allowed_headers(allowed_headers)
```

Comment:
This is more work than what we are doing in main (cloning all the request headers). We can keep a whitelist of headers or have some default "allowed-headers". In either case we should only keep a minimal set of headers in request context.